rasa/graph_components/providers/domain_provider.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+from typing import Dict, Text, Any
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import Domain
+from rasa.shared.importers.importer import TrainingDataImporter
+
+
+class DomainProvider(GraphComponent):
+    """Provides domain during training and inference time."""
+
+    def __init__(self, config: Dict[Text, Any]) -> None:
+        """Creates domain provider from config."""
+        self._config = config
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns default configuration (see parent class for full docstring)."""
+        return {
+            "remove_duplicates": True,
+            "unique_last_num_states": None,
+            "augmentation_factor": 50,
+            "tracker_limit": None,
+            "use_story_concatenation": True,
+            "debug_plots": False,
+        }
```

Comment:
Where do these config parameters come from? They don't seem to be related to domains

Reply:
😄 should have added a comment that because I wasn't sure if this component should have a config, I left these ones in from the template of `TrainingTrackerProvider` in hope to find out more during the review. I can remove this method if it's preferred to have an empty config.

Reply:
You can remove it as this one doesn't require a config 👍🏻